hal/mbed_sleep_manager.c

```diff
@@ -104,45 +74,42 @@ static void sleep_tracker_print_stats(void)
             continue;
         }
 
-        if (sleep_stats[i].identifier[0] == '\0') {
+        if (sleep_stats[i].identifier == NULL) {
             return;
         }
 
         debug("[id: %s, count: %u]\r\n", sleep_stats[i].identifier,
-                                          sleep_stats[i].count);
+                                         sleep_stats[i].count);
     }
 }
 
 void sleep_tracker_lock(const char* const filename, int line)
 {
-    const char* const stripped_path = strip_path(filename);
-
-    sleep_statistic_t* stat = sleep_tracker_find(stripped_path);
+    sleep_statistic_t *stat = sleep_tracker_find(filename);
 
     // Entry for this driver does not exist, create one.
     if (stat == NULL) {
-        stat = sleep_tracker_add(stripped_path);
+        stat = sleep_tracker_add(filename);
     }
 
     core_util_atomic_incr_u8(&stat->count, 1);
 
-    debug("LOCK: %s, ln: %i, lock count: %u\r\n", stripped_path, line, deep_sleep_lock);
+    debug("LOCK: %s, ln: %i, lock count: %u\r\n", filename, line, deep_sleep_lock);
```

Comment:
Do we really need the lock/unlock calls? I would say they just repeat the same info we get out of the who's holding sleep lock prints.

Reply:
Those changes are in a different PR: https://github.com/ARMmbed/mbed-os/pull/6349